falcon/middlewares.py

```diff
@@ -1,16 +1,80 @@
+from typing import Iterable, Optional, Union
+
+from .request import Request
+from .response import Response
+
 
 class CORSMiddleware(object):
-    def process_response(self, req, resp, resource, req_succeeded):
-        """Implement a simple blanket CORS policy for all routes.
+    """CORS Middleware
+
+    This middleware provides a simple out-of-the box CORS policy, including handling
+    of preflighted requests from the browser.
+
+    See also:
+
+    * https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+    * https://www.w3.org/TR/cors/#resource-processing-model
+
+    Keyword Arguments:
+        allow_origin (Union[str, Iterable[str]]): List of origins to allow (case
+            sensitive). The string ``'*'`` acts as a wildcard, matching every origin.
+            (default ``'*'``).
+        expose_headers (Optional[Union[str, Iterable[str]]]): List of additional headers to
```

Comment:
IMHO it might be worth adding a note that we are setting the [`Access-Control-Expose-Headers`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Expose-Headers) header, and also mention which headers are [whitelisted by default](https://developer.mozilla.org/en-US/docs/Glossary/CORS-safelisted_response_header) (if we're defining as _additional_).